velox/functions/prestosql/StringFunctions.h

```diff
@@ -507,4 +507,82 @@ struct StrLPosFunction : public StrPosFunctionBase<T, true> {};
 template <typename T>
 struct StrRPosFunction : public StrPosFunctionBase<T, false> {};
 
+// Helper method to return a byte value from the buffer.
+template <typename T>
+inline T getByte(const char* buffer, const int32_t i) {
+  return (T)buffer[i] & 0xff;
+}
+
+template <typename T>
+struct FromBigEndian32 {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE
+  void callNullFree(
+      out_type<int32_t>& result,
+      const null_free_arg_type<Varbinary>& input) {
+    static constexpr auto kTypeLength = sizeof(int32_t);
+    VELOX_USER_CHECK(
+        input.size() == kTypeLength,
+        "Expected 4-byte input, but got instead: {}",
+        input.size());
+    const char* buffer = input.data();
+    result = getByte<int32_t>(buffer, 0) << 24 |
```

Comment:
Just call `bswap_32`?

Reply:
If I am not mistaken, `bswap` is not cross-compatible and does not exist on macos.

Reply:
Maybe `__builtin_bswap32`?

Reply:
It seems I might need to do something like this: https://github.com/google/cityhash/blob/8af9b8c2b889d80c22d6bc26ba0df1afb79a30db/src/city.cc#L50 to make this cross-compatible. Is there a folly util that could do byte swap for me or shall I include this code in velox?

Reply:
We only supports gcc and clang so compiler builtin should be good

Reply:
Sorry, I think I got it all wrong, was confused by different byte order. My code was missing endianness conversion which I added + comments. We don't need `bswap` function because the first statement converts the byte array into a little endian integer. The missing part was to ensure we convert it back on big endian platforms which can be done with the `folly::Endian` set of functions.